command.go

```diff
@@ -477,6 +483,37 @@ func (c *Command) Help() error {
 	return nil
 }
 
+// SuggestFunc returns suggestions for the provided typedName using either
+// the function set by SetSuggestFunc for this command, parent's or a default one.
+// When searching for a parent's function, it recursively checks towards the root
+// and returns the first one found. If none found, uses direct parent's default.
+func (c *Command) SuggestFunc(typedName string) string {
+	if c.DisableSuggestions {
+		return ""
+	}
+	if c.suggestFunc != nil {
+		return c.suggestFunc(typedName)
+	}
+	if c.HasParent() {
+		var getParentFunc func(*Command) func(string) string
+		getParentFunc = func(parent *Command) func(string) string {
+			if parent.suggestFunc != nil {
+				return parent.suggestFunc
+			}
+			if parent.HasParent() {
+				return getParentFunc(parent.Parent())
+			}
+			return nil
```

Comment:
This might be more readable


```suggestion
			if !parent.HasParent() {
				return nil
			}
			return getParentFunc(parent.Parent())
```

Reply:
Done.